src/ctap/ctap1.rs

```diff
@@ -295,14 +294,22 @@ impl Ctap1Command {
             return Err(Ctap1StatusCode::SW_VENDOR_KEY_HANDLE_TOO_LONG);
         }
 
-        let mut response =
-            Vec::with_capacity(105 + key_handle.len() + ATTESTATION_CERTIFICATE.len());
+        let certificate = match ctap_state.persistent_store.attestation_certificate() {
+            Ok(Some(value)) => value,
+            _ => return Err(Ctap1StatusCode::SW_INS_NOT_SUPPORTED),
+        };
+        let private_key = match ctap_state.persistent_store.attestation_private_key() {
+            Ok(Some(value)) => value,
+            _ => return Err(Ctap1StatusCode::SW_INS_NOT_SUPPORTED),
+        };
```

Comment:
The `Err` and `None` are technically 2 different errors. The first is a storage error and we might want to use something vendor specific for that, to later translate it to `CTAP2_ERR_VENDOR_INTERNAL_ERROR`.

So how about
> let certificate = ctap_state.persistent_store.attestation_certificate()
 .map_err(Ctap1StatusCode::SW_VENDOR_INTERNAL_ERROR)?
.ok_or(Ctap1StatusCode::SW_INS_NOT_SUPPORTED)?;

Reply:
I don't think CTAP1 has vendor specific error codes. They are inherited from ISO7816.
I could use 0x6387 which means "Non-volatile memory is not available". WDYT?

Reply:
There is a `SW_VENDOR_KEY_HANDLE_TOO_LONG` that is currently misused also for a storage error in line 292 (currently 291). I think changing yours and that one to `0x6F00` would make most sense?

Reply:
Done for all 3.
I'm tempted to use `0x6387` instead of `SW_INS_NOT_SUPPORTED`.
First it allows us to distinguish between a key that is missing attestation crypto material and a storage failure. But also because we do support this instruction, it's just that the data is missing in the storage.

Reply:
So, I found that `0x6501` is standard code for memory failure. So I mapped storage errors to `0x6501` and missing entries to `0x6f00` (command aborted)